src/main/java/hudson/plugins/audit_trail/AuditTrailPlugin.java

```diff
@@ -24,158 +24,143 @@
 package hudson.plugins.audit_trail;
 
 import hudson.DescriptorExtensionList;
-import hudson.Plugin;
-import hudson.model.*;
-import hudson.model.Descriptor.FormException;
+import hudson.Extension;
+
+import hudson.model.AbstractBuild;
+import hudson.model.Descriptor;
+import hudson.model.Run;
 import hudson.util.FormValidation;
-import hudson.util.PluginServletFilter;
-import jenkins.model.Jenkins;
+import jenkins.model.GlobalConfiguration;
 import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
+import javax.annotation.PostConstruct;
 import javax.servlet.ServletException;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
 /**
  * Keep audit trail of particular Jenkins operations, such as configuring jobs.
+ *
  * @author Alan Harder
+ * @author Pierre Beitz
  */
-public class AuditTrailPlugin extends Plugin {
+@Symbol("audit-trail")
+@Extension
+public class AuditTrailPlugin extends GlobalConfiguration {
```

Comment:
I guess this is a breaking change? Is it not possible to avoid it?

Reply:
The jcasc documentation didn't really mention this use case.
TBH, plugin has been deprecated from quite some time now and felt while I was refactoring the plugin for JCASC I'd better also clean this. There are other breaking changes in the PR that would force scripts to be adapted anyway.

Reply:
Mmmmm not ideal I guess but not sure how it could be done in another way.

Reply:
It's possible to make it less breaking by overriding the configfile, but it's always going to break here unless you migrate the config to a new globalconfiguration...

Reply:
Ok, then it will break, it will be in the changelog and this release will bump the major :)

Reply:
@timja sorry coming back late on your latest comment...

I understand the first part of your comment and I finally decided to go with it:

```
    @Override
    protected XmlFile getConfigFile() {
        return new XmlFile(new File(Jenkins.getInstanceOrNull().getRootDir(),"audit-trail.xml"));
    }
``` 
>  but it's always going to break here unless you migrate the config to a new globalconfiguration...

But I don't understand this part. AFAICT, from an end user (not API user) point of view, the persistence will be done in `audit-trail.xml` and everybody will be happy. It's like moving a Descriptor. In which case this would break?

In case the use case I'm missing is important, to migrate to a GlobalConfiguration once and for all, I found in the doc that I'm supposed to be using a custom converter (https://wiki.jenkins.io/display/JENKINS/XStream+Tips) is this the right path?

Reply:
for an end user they won't notice anything, but if anyone else consumes the API it is a breaking change as you need to now lookup the descriptor not the plugin.

Reply:
Makes sense... as there are several other API changes anyway, it will be a breaking release for API users then. Anyway thanks a lot for your help and fast response :)